config/m4/opt.m4

```diff
@@ -11,13 +11,28 @@ AC_ARG_ENABLE(
     [opt-log],
     AC_HELP_STRING(
         [--enable-opt-log],
-        [Optimize logging output (default=no)]))
+        [Optimize logging output: none, auto, details, debug, func (default=no)]),,
+    enableval=none)
```

Comment:
The purpose of these changes is to provide a way to end user for improving performance reducing number of VMA logging information. It takes into account following assumption that messages are grouped by their  importance. So if we do not want to see `debug` messages then all messages starting `debug` as `func`, `funcall` should be disabled too.
So these methodology does not cover scenarios described in question.